### ISSUE TITLE
variable behavior with  CS 7.0

### ISSUE THREAD

**Issue by ludan-av** [bwilcox-1234/ChatScript#30]
$_dataRaw is a variable that is already set.
Considering this code:

```
# printing $_dataRaw
BEFORE: $_dataRaw

# setting another variable
$_dataX = "hello"

# printing $_dataRaw
AFTER: $_dataRaw
```

With CS 6.9, the output is:
`BEFORE: { "firstName": "André" }  AFTER: { "firstName": "André" }`
(which seemed ok to me)

With CS 7.0, the output is: 
`BEFORE: { "firstName": "André" }  AFTER: "hello"`
(which seems wrong)

**Comment by bwilcox-1234** [comment]
I'll need more data. This example works fine:
topic: ~introductions  keep repeat()
t: hi there 
u:  (test ) $_x =  ^' { "firstName": "André" } ' 
	$_x
	$_dataX = "hello"
	$_x

**Comment by ludan-av** [comment]
Sure.
Below is the minimalist code to produce the issue:

```
topic: ~debut repeat []

t: choose Andre ^keep()

u: (Andre)
  $_dataRaw = ^'
  {
    "civilite": "M.",
    "firstName": "André"
  }
  '

u: ($_dataRaw)  

  # printing $_dataRaw
  BEFORE: $_dataRaw
  
  # setting another variable
  $_dataX = "hello"
  
  # printing $_dataRaw
  AFTER: $_dataRaw
```

The output is:
```
BEFORE:  { "civilite": "M.", "firstName": "André" }  AFTER: "hello"
```

**Comment by bwilcox-1234** [comment]
thank you for your bug report. Fix unloaded as Version 7.01.  The bug occurred with local variables in topics crossing rule boundaries.
